clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1885,6 +1886,24 @@ public void testUnsubscribeWithoutGroupId() {
         verify(applicationEventHandler).add(ArgumentMatchers.isA(UnsubscribeEvent.class));
     }
 
+    @Test
+    public void testOffsetResetEvent() {
+        SubscriptionState subscriptions = mock(SubscriptionState.class);
+        TopicPartition topic = new TopicPartition("test", 0);
+        consumer = spy(newConsumer(
+                mock(FetchBuffer.class),
+                new ConsumerInterceptors<>(Collections.emptyList()),
+                mock(ConsumerRebalanceListenerInvoker.class),
+                subscriptions,
+                "group-id",
+                "client-id"));
+        consumer.seekToBeginning(Collections.singleton(topic));
+        verify(applicationEventHandler).add(any(ResetOffsetEvent.class));
+        verify(subscriptions, never()).requestOffsetReset(topic);
+        completeResetOffsetEventSuccessfully(topic, subscriptions);
+        verify(subscriptions).requestOffsetReset(topic);
```

Comment:
Although the test case passes, I think the statement here is wrong. When using `seekToBeginning`, it should call `SubscriptionState#requestOffsetReset(topic, OffsetResetStrategy.EARLIEST)`. We can change `completeResetOffsetEventSuccessfully` function like following and update the statement to `verify(subscriptions).requestOffsetReset(topic, OffsetResetStrategy.EARLIEST);` here:

```java
    private void completeResetOffsetEventSuccessfully(TopicPartition tp, SubscriptionState subscriptions) {
        doAnswer(invocation -> {
            ResetOffsetEvent event = invocation.getArgument(0);
            consumer.subscriptions().requestOffsetReset(event.topicPartitions(), event.offsetResetStrategy());
            return null;
        }).when(applicationEventHandler).add(ArgumentMatchers.isA(ResetOffsetEvent.class));
    }
```

You can also add another test for `seekToEnd` as well.